lib/rouge/lexers/svelte.rb

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Svelte < HTML
+      desc 'Svelte single-file components (https://svelte.dev/)'
+      tag 'svelte'
+      filenames '*.svelte'
+
+      # are these accurate?
+      mimetypes 'text/x-svelte', 'application/x-svelte'
+
+      def initialize(*)
+        super
+        # todo add support for typescript script blocks
+        @js = Javascript.new(options)
+      end
+
+      # Shorthand syntax for passing attributes - ex, `{src}` instead of `src={src}`
+      prepend :tag do
+        rule %r/(\{)\s*([a-zA-Z0-9_]+)\s*(})/m do
+          groups Str::Interpol, Name::Variable, Str::Interpol
+          pop!
+        end
+      end
+
+      prepend :attr do
+        # Duplicate template_start mixin here with a pop!
+        # Because otherwise we'll never exit the attr state
+        rule %r/\{/ do
+          token Str::Interpol
+          pop!
+          push :template
+        end
+      end
+
+      # handle templates within attribute single/double quotes
+      prepend :dq do
+        mixin :template_start
+      end
+
+      prepend :sq do
+        mixin :template_start
+      end
+
+      prepend :root do
+        # detect curly braces within HTML text (outside of tags/attributes)
+        rule %r/([^<&{]*)(\{)(\s*)/ do
+          groups Text, Str::Interpol, Text
+          push :template
+        end
+      end
+
+
+      state :template_start do
```

Comment:
Remove excessive lines.

```suggestion
      state :template_start do
```